services/user/Supervisor/ui/src/oauth/App.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from "react";
+import { siblingUrl } from "@psibase/common-lib";
+import { ActiveOauthRequest, isTypeOauthRequest, OauthRequest } from "./db";
+
+const buildIframeUrl = (promptUserReq: OauthRequest) => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const url = siblingUrl(
+        null,
+        promptUserReq.subdomain,
+        promptUserReq.subpath,
+        true,
+    );
+    const newIframeUrl = new URL(url);
+    newIframeUrl.searchParams.set("id", promptUserReq.id);
+    newIframeUrl.searchParams.set(
+        "returnUrlPath",
+        urlParams.get("returnUrlPath") || "",
+    );
+    return newIframeUrl.toString();
+};
+
+export const App = () => {
+    const [iframeUrl, setIframeUrl] = useState<string | null>(null);
+    const [error, setError] = useState<string | null>(null);
+
+    const initApp = async () => {
```

Comment:
Same here. I'd wrap `initApp` in a `useCallback()` so it's not re-created on each render (unless its dependencies change). Or alternatively, move it into the `useEffect()` right above where you call it. That would have a similar result.